internal/urlutil/url.go

```diff
@@ -77,3 +78,26 @@ func GetAbsoluteURL(r *http.Request) *url.URL {
 	u.Host = r.Host
 	return u
 }
+
+// GetDomainsForURL returns the available domains for given url.
```

Comment:
@cuonglm could you add some tests for this function?

Reply:
Yes, especially since the ordering of the returned slice matters.